### DISCUSSION TITLE
Getting queries out of SQLModel

### DISCUSSION THREAD

**Discussion by jochman** [fastapi/sqlmodel#809] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional


from sqlmodel import SQLModel

class Model(SQLModel, table=true):
  id: Optional[int] = Field(default=None, primary_key=True)
  value: str
```

### Description

I need to get the query text of creating this table - how do I do that without creating a new sqlalchemy object to replace this?

### Operating System

Linux

### SQLModel Version

0.0.14

### Python Version

3.12.2

**Comment by riziles** [comment, marked as answer]
@jochman , Try this:

```python
from sqlmodel import Field, SQLModel, create_engine
from sqlalchemy.schema import CreateTable

class Hero(SQLModel, table=True):
    id: int = Field(default=None, primary_key=True)
    name: str
    secret_name: str

sqlite_file_name = "database.db"
sqlite_url = f"sqlite:///{sqlite_file_name}"

engine = create_engine(sqlite_url, echo=True)

print(CreateTable(Hero.__table__).compile(engine))
```

found it here: https://stackoverflow.com/questions/2128717/sqlalchemy-printing-raw-sql-from-create
